examples/append-to-dom/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+use percy_dom::{VirtualNode, JsCast, event::VirtualEvents, html};
+use web_sys::{HtmlElement, HtmlInputElement};
+use wasm_bindgen_test::*;
+
+wasm_bindgen_test_configure!(run_in_browser);
+
+#[wasm_bindgen_test]
+pub fn main() {
```

Comment:
Let's make sure to demonstrate the following:

- Doing all of this using the `html!` macro
- Doing all of this without using the `html!` macro
- Using `.special_attributes.on_create_element` to demonstrate how to ensure that the unmanaged element gets inserted after `percy` creates the DOM node.
  - useful when you aren't appending to the root node since you won't have a `div_elem: web_sys::Element` on hand like you do in this example

Let's make the example append the input to a non-root node. This is the more common scenario. Or, feel free to demonstrate both.

I'm imagining that this `fn main` can call some different functions that each demonstrate some subset of the above list + the things that you are already demonstrating.

Reply:
> Let's make the example append the input to a non-root node

Turns out this forces a mix of the above points such that there's only really one good way of doing it... sort of. Not using `html!` at all is possible and isn't demonstrated, but I don't think showing the example without `html!` is adding anything but verbosity, as `html!` currently isn't being used for anything that can't be easily manually constructed with `VElement`s.

I can remove the use of `html!` if you feel strongly about it, or alternatively add another version of the function that doesn't use `html!`.

I recommend checking the new example first and seeing if you feel its necessary, once I commit it. (I'm not exactly sure why you want two versions.)

Reply:
Only using `html!` sounds good.

Agree that we shouldn't mix concepts unless it truly demonstrates something important.

We can always have a future example that shows how to create nodes with the macro.